test/distributed/test_inductor_collectives.py

```diff
@@ -136,7 +212,8 @@ def func(inp, *, tag, ranks, group_size):
             .check("buf0 = empty_strided") \
             .check("buf0.copy_(arg0_1)") \
             .check("buf0_work = dist.all_reduce(buf0") \
-            .check("buf0_work.wait()") \
+            .check("_register_tensor_work(buf0, buf0_work)") \
```

Comment:
We should later add comments to _functional_collectives mentioning that those methods are used by inductor